sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -64,19 +65,36 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                 throw new InvalidOperationException("Bearer token authentication is not permitted for non TLS protected (https) endpoints.");
             }
 
-            if (DateTimeOffset.UtcNow >= _refreshOn)
+            (string? headerValue, Task<string>? pending) = _accessTokenCache.UpdateTokenState();
+            if (headerValue == default)
             {
-                AccessToken token = async ?
-                        await _credential.GetTokenAsync(new TokenRequestContext(_scopes, message.Request.ClientRequestId), message.CancellationToken).ConfigureAwait(false) :
-                        _credential.GetToken(new TokenRequestContext(_scopes, message.Request.ClientRequestId), message.CancellationToken);
+                if (pending != null)
+                {
+                    headerValue = async
+                        ? await pending.WithCancellation(message.CancellationToken)
+                        : pending.WaitWithCancellation(message.CancellationToken);
```

Comment:
BTW, this is an easy way to get a thread pool during peak loads.

Reply:
I'm not sure I understand how is it different from any other synchronous call that holds a thread in thread pool that we have.

Reply:
I guess you are right it doesn't make things worse. It just looks like a chokepoint similar to the ones that I've seen causing issues in the past.